sqlglot/transforms.py

```diff
@@ -80,4 +80,24 @@ def _transform(self, *args, **kwargs):
     return _transform
 
 
+def add_or_dpipe(expression):
+    if isinstance(expression, exp.Add) and expression.type in (
+        exp.DataType.Type.VARCHAR,
+        exp.DataType.Type.NVARCHAR,
+        exp.DataType.Type.CHAR,
+        exp.DataType.Type.NCHAR,
+        exp.DataType.Type.TEXT,
+    ):
+        expression = exp.DPipe(this=expression.this, expression=expression.expression)
```

Comment:
Which is more standard, `||` or `CONCAT`?

Reply:
probably concat